starters/express-typescript-typeorm-postgres/src/controllers/technology/handlers/delete-technology.ts

```diff
@@ -1,22 +1,49 @@
 import { NextFunction, Request, Response } from 'express';
 import { StatusCodes } from 'http-status-codes';
+import { clearCacheEntry } from '../../../cache/cache';
+import { Result } from '../../../constants/result';
 import { dataSource } from '../../../db/datasource';
 import { Technology } from '../../../entities/technology.entity';
+import { ErrorResult, SuccessResult } from '../../../interfaces/results';
+import { LogHelper } from '../../../utils/log-helper';
+
+type DeleteTechnologyResult = SuccessResult<null> | ErrorResult;
 
 export async function deleteTechnology(
   req: Request,
   res: Response,
   next: NextFunction
 ): Promise<void> {
-  try {
-    const technologyId: number = parseInt(req.params.technologyId);
-    await dataSource.getRepository(Technology).delete({
-      id: technologyId,
-    });
-    res.status(StatusCodes.OK).json({
-      id: technologyId,
-    });
-  } catch (e) {
-    next(e);
+  const technologyId: number = parseInt(req.params.technologyId);
+  const deleteResult = await deleteTechnologyEntry(technologyId);
+
+  if (deleteResult.type === Result.ERROR) {
+    LogHelper.error(deleteResult.message, deleteResult.error);
+    next(deleteResult.error);
+    return;
   }
+
+  clearCacheEntry(req.baseUrl);
+  clearCacheEntry(req.originalUrl);
+
+  res.status(StatusCodes.OK).json({
+    id: technologyId,
+  });
+}
+
+function deleteTechnologyEntry(technologyId: number): Promise<DeleteTechnologyResult> {
```

Comment:
I agree on extracting the methods, I'm not sure about abstracting it into a standard interface. Do we want to be that opinionated? Let's get back to this when you are online.

Reply:
> I agree on extracting the methods
Just do that.  I get overly excited about over-engineering sometimes. I still think it could be a good change, but we don't have to do it here

Reply:
I extracted them, I'll resolve this conversation now